pkg/mount/block/blockdev.go

```diff
@@ -165,6 +170,42 @@ func (b *BlockDev) ReadPartitionTable() error {
 	return unix.IoctlSetInt(int(f.Fd()), unix.BLKRRPART, 0)
 }
 
+// FindPCIInfo searches sysfs for the PCI vendor and device id.
+// We fill in the PCI struct with just those two elements.
+func (b *BlockDev) FindPCIInfo() (*pci.PCI, error) {
+	dPath := filepath.Join("/sys/class/block", b.Name, "device")
+	// Loop through devices until we find the actual backing pci device.
+	d, err := os.Stat(dPath)
+	for err == nil && d.IsDir() {
+		dPath = filepath.Join(dPath, "device")
+		d, err = os.Stat(dPath)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	// At this point we have a file named device that is not a directory.
+	// Search for a vendor file in the same directory.
+	vPath := filepath.Join(filepath.Dir(dPath), "vendor")
+	if v, err := os.Stat(vPath); err != nil {
+		return nil, err
+	} else if v.IsDir() {
+		return nil, fmt.Errorf("vendor path %v is a directory, need file", vPath)
+	}
+
+	// Read both files into the pci struct and return
+	device, err := ioutil.ReadFile(dPath)
+	if err != nil {
+		return nil, fmt.Errorf("Error reading device file: %v", err)
+	}
+	vendor, err := ioutil.ReadFile(vPath)
+	if err != nil {
+		return nil, fmt.Errorf("Error reading vendor file: %v", err)
+	}
+	return &pci.PCI{Vendor: strings.TrimSpace(string(vendor)), Device: strings.TrimSpace(string(device)),
```

Comment:
can you make this multiline?

```go
return &pci.PCI{
  Vendor: ,
  Device: ,
  FullPath: ,
}, nil
```

Reply:
whereas Ron's reaction is: I like single line, I can just grep and see the whole thing.

Reply:
Changed to multiline, sorry ron,